adapters/Maio/MaioAdapter/GADMAdapterMaioRewardedAd.m

```diff
@@ -55,131 +55,65 @@ - (void)loadRewardedAdForAdConfiguration:(GADMediationRewardedAdConfiguration *)
   _zoneId = adConfiguration.credentials.settings[GADMMaioAdapterZoneIdKey];
 
-  if (!self.mediaId) {
-    NSError *error = GADMAdapterMaioErrorWithCodeAndDescription(
-        GADMAdapterMaioErrorInvalidServerParameters,
-        @"maio mediation configurations did not contain a valid media ID.");
-    completionHandler(nil, error);
-    return;
-  }
-
-  GADMAdapterMaioAdsManager *adManager =
-      [GADMAdapterMaioAdsManager getMaioAdsManagerByMediaId:_mediaId];
-
-  GADMAdapterMaioRewardedAd *__weak weakSelf = self;
-  [adManager initializeMaioSDKWithCompletionHandler:^(NSError *error) {
-    if (error) {
-      self.completionHandler(nil, error);
-    } else {
-      // 生成済みのinstanceを得た場合、testモードを上書きする必要がある
-      [adManager setAdTestMode:adConfiguration.isTestRequest];
-      NSError *error = [adManager loadAdForZoneId:weakSelf.zoneId delegate:self];
-      if (error) {
-        self.completionHandler(nil, error);
-      }
-    }
-  }];
+  MaioRequest *request = [[MaioRequest alloc] initWithZoneId:self.zoneId testMode:adConfiguration.isTestRequest];
+  self.rewarded = [MaioRewarded loadAdWithRequest:request callback:self];
 }
 
 - (void)presentFromViewController:(nonnull UIViewController *)viewController {
-  GADMAdapterMaioAdsManager *adManager =
-      [GADMAdapterMaioAdsManager getMaioAdsManagerByMediaId:_mediaId];
-
-  [self.adEventDelegate willPresentFullScreenView];
-  [adManager showAdForZoneId:self.zoneId rootViewController:viewController];
+  [self.rewarded showWithViewContext:viewController callback:self];
 }
 
-#pragma mark - MaioDelegate
+#pragma mark - MaioRewardedLoadCallback, MaioRewardedShowCallback
 
-/**
- *  全てのゾーンの広告表示準備が完了したら呼ばれます。
- */
-- (void)maioDidInitialize {
-  // noop
+- (void)didLoad:(MaioRewarded *)ad {
+  self.adEventDelegate = self.completionHandler(self, nil);
 }
 
-/**
- *  広告の配信可能状態が変更されたら呼ばれます。
- *
- *  @param zoneId   広告の配信可能状態が変更されたゾーンの識別子
- *  @param newValue 変更後のゾーンの状態。YES なら配信可能
- */
-- (void)maioDidChangeCanShow:(NSString *)zoneId newValue:(BOOL)newValue {
-  if (!newValue) {
-    return;
-  }
-  id<GADMediationRewardedAdEventDelegate> delegate = self.completionHandler(self, nil);
-  // Maio SDK may call maioDidChangeCanShow multiple times during the lifecycle of the ad. Avoid
-  // overwriting the delegate to nil since the completion handler only returns a non-nil delegate
-  // once.
-  if (delegate) {
-    self.adEventDelegate = delegate;
+- (void)didFail:(MaioRewarded *)ad errorCode:(NSInteger)errorCode {
+  NSString *description = @"maio SDK returned error";
+  NSDictionary *userInfo = @{
+    NSLocalizedDescriptionKey: description,
+    NSLocalizedFailureReasonErrorKey: description
+  };
+  NSError *error = [NSError errorWithDomain:GADMMaioSDKErrorDomain
+                                       code:errorCode
+                                   userInfo:userInfo];
+
+  NSLog(@"Maiorewarded did fail. error: %@", error);
```

Comment:
Update the logging message to: `Maio rewarded ad failed to load with error code: %@`